src/cast_core/src/RollingCollections.cpp

```diff
@@ -260,6 +253,65 @@ struct Manage : public RunOperation {
     std::vector<DocumentGenerator> _indexConfig;
 };
 
+struct OplogTailer : public RunOperation {
+    OplogTailer(PhaseContext& phaseContext,
+                mongocxx::database db,
+                ActorId id,
+                RollingCollectionNames& rollingCollectionNames)
+        : RunOperation(db, rollingCollectionNames),
+          _cursor{},
+          _oplogLagOperation{phaseContext.operation("OplogLag", id)} {
+        if (phaseContext.actor()["Threads"].to<int>() != 1) {
+            BOOST_THROW_EXCEPTION(
+                InvalidConfigurationException("OplogTailer can only be run with one thread"));
+        }
+    }
+
+    void run() override {
+        if (_firstLoop) {
+            mongocxx::options::find opts{};
+            opts.cursor_type(mongocxx::cursor::type::k_tailable);
+            _cursor = std::optional<mongocxx::cursor>(database["oplog.rs"].find({}, opts));
+            /*
+             * Exhaust the cursor to skip the initially created collections,
+             * i.e. the admin / local tables and the intial set of rolling
+             * collections, this avoids a spike in latency.
+             */
+            for (auto&& doc : _cursor.value()) {
+                // Do nothing.
+            }
+            _firstLoop = false;
+        }
+        for (auto&& doc : _cursor.value()) {
+            if (doc["op"].get_utf8().value.to_string() == "c") {
+                auto object = doc["o"].get_document().value;
+                auto it = object.find("create");
+                if (it != object.end()) {
```

Comment:
consider negating this and `continue`ing to save some indentation

```c++
if (it == end()) { continue; }
```

same also applies to above

```c++
if (doc["op"].get_utf8().value.to_string() != "c") { continue; }
```

Reply:
I dunno I kinda like it as is.